### PR TITLE
Feature: flyway

### PR DESCRIPTION
**Pull request by doncem**
### JIRA link (if applicable) ###

[Create metadata records in database](https://tools.hmcts.net/jira/browse/RPE-554)

### Change description ###

Initial setup of database and flyway

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: config/owasp/suppressions.xml [hunk 1]
```diff
@@ -1,5 +1,20 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <suppressions xmlns="https://jeremylong.github.io/DependencyCheck/dependency-suppression.1.1.xsd">
+
+  <suppress>
+    <notes><![CDATA[postgresql before versions 10.4, 9.6.9 is vulnerable in the adminpack extension, the pg_catalog.pg_logfile_rotate() function doesn't follow the same ACLs than pg_rorate_logfile. If the adminpack is added to a database, an attacker able to connect to it could exploit this to force log rotation]]></notes>
```
Comment (rkondratowicz) [inline config/owasp/suppressions.xml]: a comment on why this got suppressed would be nice

Comment (doncem) [reply config/owasp/suppressions.xml]: the suppression itself explains it in detail: notes text copy from web + gav regex to which dependency is a culprit + cve - link to detailed info

File: build.gradle [hunk 2]
```diff
@@ -147,6 +148,11 @@ dependencies {
   compile group: 'org.springframework.boot', name: 'spring-boot-starter-actuator', version: versions.springBoot
   compile group: 'org.springframework.boot', name: 'spring-boot-starter-aop', version: versions.springBoot
   compile group: 'org.springframework.boot', name: 'spring-boot-starter-json', version: versions.springBoot
+  compile group: 'org.springframework.boot', name: 'spring-boot-starter-data-jpa', version: versions.springBoot
+
+  compile group: 'org.flywaydb', name: 'flyway-core', version: '5.1.3'
+  compile group: 'org.postgresql', name: 'postgresql', version: '42.2.2'
+  compile group: 'com.vladmihalcea', name: 'hibernate-types-52', version: '2.2.2'
```
[the remark this answers is not in the document]
Comment (doncem) [reply build.gradle]: Not relevant but can be reviewed with actual db setup. I should a comment to review afterwords 👍

File: build.gradle [hunk 3]
```diff
@@ -161,6 +167,7 @@ dependencies {
 
   integrationTestCompile sourceSets.main.runtimeClasspath
   integrationTestCompile sourceSets.test.runtimeClasspath
+  integrationTestCompile group: 'org.testcontainers', name: 'postgresql', version: '1.8.0'
```
[the remark this answers is not in the document]
Comment (doncem) [reply build.gradle]: This is needed for integration tests to have postgresql db running

File: src/integrationTest/resources/testcontainers.properties [hunk 4]
```diff
@@ -0,0 +1 @@
+checks.disable=true
```
[the remark this answers is not in the document]
Comment (doncem) [reply src/integrationTest/resources/testcontainers.properties]: Speeds up tests significantly

File: config/owasp/suppressions.xml (hunk 1 above)
Comment (rkondratowicz) [reply config/owasp/suppressions.xml]: It explains why there is a problem with this library, not why you've decided to ignore this warning.

**Pull request merged**